
# Data import 


```{r, include = FALSE}
ottr::set_knitr_image_path()
```

## Learning Objectives

- Import read depth signal
- {Another learning objective}

# Import read depth signal

Make sure that you have indexed alignment (SAM, BAM or CRAM) file.
Initialize your CNVpytor project by running:
```
> cnvpytor -root file.pytor -rd file.bam [-chrom name1 ...] [-T ref.fa.gz]
```
where:

* file.pytor -- specifies output cnvpytor file (HDF5 file)
* name1 ... -- specifies chromosome name(s).
* file.bam -- specifies bam/sam/cram file name.
* -T ref.fa.gz -- specifies reference genome file (only for cram file without reference genome).


Chromosome names must be specified the same way as they are described in the sam/bam/cram header, 
e.g., chrX or X. One can specify multiple chromosomes separated by space. If no chromosome is specified, 
read mapping is extracted for all chromosomes in the sam/bam file. 
Note that the pytor file is not being overwritten.

**Examples:**

```
> cnvpytor -root NA12878.pytor -chrom 1 2 3 -rd NA12878_ali.bam
```
for bam files with a header like this:
```
@HD VN:1.4 GO:none SO:coordinate
@SQ SN:1 LN:249250621
@SQ SN:2 LN:243199373
@SQ SN:3 LN:198022430
```

or
```
> cnvpytor -root NA12878.pytor -chrom chr1 chr2 chr3 -rd NA12878_ali.bam
```
for bam files with a header like this:
```
@HD VN:1.4 GO:none SO:coordinate
@SQ SN:chr1 LN:249250621
@SQ SN:chr2 LN:243199373
@SQ SN:chr3 LN:198022430
```


After -rd step file file.pytor is created and read depth data binned to 100 base pair bins will be stored 
in _pytor_ file.

Chromosome names and lengths are parsed from the input file header and used to 
detect reference genome.

## Subtopic

Here's a subheading and some text in this subsection!


