<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Installation and Setting Reference genome | Copy Number Analysis using CNVpytor</title>
  <meta name="description" content="Chapter 2 Installation and Setting Reference genome | Copy Number Analysis using CNVpytor" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Installation and Setting Reference genome | Copy Number Analysis using CNVpytor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Chapter 2 Installation and Setting Reference genome | Copy Number Analysis using CNVpytor" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Installation and Setting Reference genome | Copy Number Analysis using CNVpytor" />
  
  <meta name="twitter:description" content="Chapter 2 Installation and Setting Reference genome | Copy Number Analysis using CNVpytor" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/dasl_favicon.ico" type="image/x-icon" />
<link rel="prev" href="introduction.html"/>
<link rel="next" href="calling-cnv-from-alignment-file-using-terminal.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Course</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html"><i class="fa fa-check"></i><b>2</b> Installation and Setting Reference genome</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#learning-objectives"><i class="fa fa-check"></i><b>2.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#libraries"><i class="fa fa-check"></i><b>2.2</b> Libraries</a></li>
<li class="chapter" data-level="2.3" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#installation"><i class="fa fa-check"></i><b>2.3</b> Installation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#install-by-cloning-from-github"><i class="fa fa-check"></i><b>2.3.1</b> Install by cloning from GitHub</a></li>
<li class="chapter" data-level="2.3.2" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#install-using-pip"><i class="fa fa-check"></i><b>2.3.2</b> Install using pip</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#steps-for-setting-reference-genome"><i class="fa fa-check"></i><b>2.4</b> Steps for setting reference genome</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#create-gc-and-mask-file-for-new-reference-genome"><i class="fa fa-check"></i><b>2.4.1</b> Create GC and mask file for new reference genome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html"><i class="fa fa-check"></i><b>3</b> Calling CNV from Alignment file using terminal</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="3.2" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html#calling-cnv-from-alignment-file"><i class="fa fa-check"></i><b>3.2</b> Calling CNV from Alignment file</a></li>
<li class="chapter" data-level="3.3" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html#importing-and-using-variant-information"><i class="fa fa-check"></i><b>3.3</b> Importing and using variant information:</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-import.html"><a href="data-import.html"><i class="fa fa-check"></i><b>4</b> Data import</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-import.html"><a href="data-import.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.1</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html"><i class="fa fa-check"></i><b>5</b> Import read depth signal</a>
<ul>
<li class="chapter" data-level="5.1" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#check-reference-genome"><i class="fa fa-check"></i><b>5.1</b> check reference genome</a></li>
<li class="chapter" data-level="5.2" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#import-snp-data"><i class="fa fa-check"></i><b>5.2</b> Import SNP data</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#from-variant-file"><i class="fa fa-check"></i><b>5.2.1</b> From variant file</a></li>
<li class="chapter" data-level="5.2.2" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#using-snp-positions-from-variant-file-and-counts-from-alignment-file"><i class="fa fa-check"></i><b>5.2.2</b> Using SNP positions from variant file and counts from alignment file</a></li>
<li class="chapter" data-level="5.2.3" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#calculating-baf-histograms"><i class="fa fa-check"></i><b>5.2.3</b> Calculating BAF histograms</a></li>
<li class="chapter" data-level="5.2.4" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#predicting-cnv-regions-using-joint-caller-prototype"><i class="fa fa-check"></i><b>5.2.4</b> Predicting CNV regions using joint caller (prototype)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#genotyping-genomic-regions"><i class="fa fa-check"></i><b>5.3</b> Genotyping genomic regions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="a-new-chapter.html"><a href="a-new-chapter.html"><i class="fa fa-check"></i><b>6</b> A new chapter</a>
<ul>
<li class="chapter" data-level="6.1" data-path="a-new-chapter.html"><a href="a-new-chapter.html#learning-objectives-3"><i class="fa fa-check"></i><b>6.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="6.2" data-path="a-new-chapter.html"><a href="a-new-chapter.html#libraries-1"><i class="fa fa-check"></i><b>6.2</b> Libraries</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="topic-of-section.html"><a href="topic-of-section.html"><i class="fa fa-check"></i><b>7</b> Topic of Section</a>
<ul>
<li class="chapter" data-level="7.1" data-path="topic-of-section.html"><a href="topic-of-section.html#subtopic"><i class="fa fa-check"></i><b>7.1</b> Subtopic</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="a-new-chapter-1.html"><a href="a-new-chapter-1.html"><i class="fa fa-check"></i><b>8</b> A new chapter</a>
<ul>
<li class="chapter" data-level="8.1" data-path="a-new-chapter-1.html"><a href="a-new-chapter-1.html#learning-objectives-4"><i class="fa fa-check"></i><b>8.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="8.2" data-path="a-new-chapter-1.html"><a href="a-new-chapter-1.html#libraries-2"><i class="fa fa-check"></i><b>8.2</b> Libraries</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html"><i class="fa fa-check"></i><b>9</b> Topic of Section</a>
<ul>
<li class="chapter" data-level="9.1" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#subtopic-1"><i class="fa fa-check"></i><b>9.1</b> Subtopic</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#code-examples"><i class="fa fa-check"></i><b>9.1.1</b> Code examples</a></li>
<li class="chapter" data-level="9.1.2" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#image-example"><i class="fa fa-check"></i><b>9.1.2</b> Image example</a></li>
<li class="chapter" data-level="9.1.3" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#video-examples"><i class="fa fa-check"></i><b>9.1.3</b> Video examples</a></li>
<li class="chapter" data-level="9.1.4" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#links-to-files"><i class="fa fa-check"></i><b>9.1.4</b> Links to files</a></li>
<li class="chapter" data-level="9.1.5" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#links-to-websites"><i class="fa fa-check"></i><b>9.1.5</b> Links to websites</a></li>
<li class="chapter" data-level="9.1.6" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#citation-examples"><i class="fa fa-check"></i><b>9.1.6</b> Citation examples</a></li>
<li class="chapter" data-level="9.1.7" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#fyi-boxes"><i class="fa fa-check"></i><b>9.1.7</b> FYI boxes</a></li>
<li class="chapter" data-level="9.1.8" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#dropdown-summaries"><i class="fa fa-check"></i><b>9.1.8</b> Dropdown summaries</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="topic-of-section-1.html"><a href="topic-of-section-1.html#print-out-session-info"><i class="fa fa-check"></i><b>9.2</b> Print out session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by:</a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Copy Number Analysis using CNVpytor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=1.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/style.css" rel="stylesheet">
</head>



<div class="hero-image-container">
  <img class= "hero-image" src= "https://github.com/jhudsl/OTTR_Template/raw/main/assets/dasl_thin_main_image.png">
</div>
<div id="installation-and-setting-reference-genome" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Installation and Setting Reference genome</h1>
<div id="learning-objectives" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Learning Objectives</h2>
<p>This chapter will cover:</p>
<ul>
<li>Installation
<ul>
<li>using setuptools</li>
<li>using pip</li>
</ul></li>
<li>Steps for setting reference genome
<ul>
<li>Create GC and mask file for new reference genome</li>
</ul></li>
</ul>
</div>
<div id="libraries" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Libraries</h2>
<p>CNVpytor is written in python and it works on both python 2 and 3. Please install python before proceeding with the installation steps.</p>
<p>The following code can be used to check python version</p>
<pre><code>python --version</code></pre>
</div>
<div id="installation" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Installation</h2>
<div id="install-by-cloning-from-github" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Install by cloning from GitHub</h3>
<p>The following lines of codes can be used to install directly from GitHub</p>
<pre><code>&gt; git clone https://github.com/abyzovlab/CNVpytor.git
&gt; cd CNVpytor
&gt; pip install .</code></pre>
<p>For single user (without admin privileges), can use:</p>
<pre><code>&gt; pip install --user .</code></pre>
</div>
<div id="install-using-pip" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Install using pip</h3>
<p>The following code will download the latest code from GitHub and use pip to install.</p>
<pre><code>pip install git+https://github.com/abyzovlab/CNVpytor.git</code></pre>
</div>
</div>
<div id="steps-for-setting-reference-genome" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Steps for setting reference genome</h2>
<p>Commonly used human reference genomes (HG19 and HG38) are integrated and comes with default GitHub installation. Its detects the genome by comparing the chromosome lengths. Although, other reference genomes are also frequently used in practice. This section will guide one to add a new reference genome.</p>
<p>Now, we will create example configuration file for mouse reference genome <code>MGSCv37</code> which contains a list of chromosomes and chromosome lengths:</p>
<pre><code># Filename: example_ref_genome_conf.py

import_reference_genomes = {
    &quot;mm9&quot;: {
        &quot;name&quot;: &quot;MGSCv37&quot;,
        &quot;species&quot;: &quot;Mus musculus&quot;,
        &quot;chromosomes&quot;: OrderedDict(
            [(&quot;chr1&quot;, (197195432, &quot;A&quot;)), (&quot;chr2&quot;, (181748087, &quot;A&quot;)), (&quot;chr3&quot;, (159599783, &quot;A&quot;)),
            (&quot;chr4&quot;, (155630120, &quot;A&quot;)), (&quot;chr5&quot;, (152537259, &quot;A&quot;)), (&quot;chr6&quot;, (149517037, &quot;A&quot;)),
            (&quot;chr7&quot;, (152524553, &quot;A&quot;)), (&quot;chr8&quot;, (131738871, &quot;A&quot;)), (&quot;chr9&quot;, (124076172, &quot;A&quot;)),
            (&quot;chr10&quot;, (129993255, &quot;A&quot;)), (&quot;chr11&quot;, (121843856, &quot;A&quot;)), (&quot;chr12&quot;, (121257530, &quot;A&quot;)),
            (&quot;chr13&quot;, (120284312, &quot;A&quot;)), (&quot;chr14&quot;, (125194864, &quot;A&quot;)), (&quot;chr15&quot;, (103494974, &quot;A&quot;)),
            (&quot;chr16&quot;, (98319150, &quot;A&quot;)), (&quot;chr17&quot;, (95272651, &quot;A&quot;)), (&quot;chr18&quot;, (90772031, &quot;A&quot;)),
            (&quot;chr19&quot;, (61342430, &quot;A&quot;)), (&quot;chrX&quot;, (166650296, &quot;S&quot;)), (&quot;chrY&quot;, (15902555, &quot;S&quot;)),
            (&quot;chrM&quot;, (16299, &quot;M&quot;))]),
        &quot;gc_file&quot;: &quot;/..PATH../MGSCv37_gc_file.pytor&quot;,
        &quot;mask_file&quot;: &quot;/..PATH../MGSCv37_mask_file.pytor&quot;
    }
}</code></pre>
<p>Letter next to chromosome length denote type of a chromosome: A - autosome, S - sex chromosome, M - mitochondria. The instruction for creating <code>MGSCv37_gc_file.pytor</code> and <code>MGSCv37_mask_file.pytor</code> is in the next section.</p>
<p>To use CNVpytor with new reference genome us <code>-conf</code> option in each <code>cnvpytor</code> command, e.g.</p>
<pre><code>cnvpytor -conf REL_PATH/example_ref_genome_conf.py -root file.pytor -rd file.bam</code></pre>
<p>CNVpytor will use chromosome lengths from alignment file to detect reference genome. However, if you configured reference genome after you had already run -rd step you could assign reference genome using <code>-rg</code>:</p>
<pre><code>cnvpytor -conf REL_PATH/example_ref_genome_conf.py -root file.pytor -rg mm9</code></pre>
<p>To avoid typing <code>-conf REL_PATH/example_ref_genome_conf.py</code> each time you run <code>cnvpytor</code>, you can create an bash alias or make configuration permanent by copying <code>example_ref_genome_conf.py</code> to <code>~/.cnvpytor/reference_genomes_conf.py</code>.</p>
<div id="create-gc-and-mask-file-for-new-reference-genome" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Create GC and mask file for new reference genome</h3>
<p>CNVpytor also has optional features for GC correction and masking (i.e., commonly known false positive regions). One can setup their reference genome by adding its related content in the <code>gc_file</code> and <code>mask_file</code> field of the configuration file.</p>
<p>To create <code>GC</code> file, we need sequence of the reference genome in <code>fasta.gz</code> file:</p>
<pre><code>&gt; cnvpytor -root MGSCv37_gc_file.pytor -gc ~/hg19/mouse.fasta.gz -make_gc_file</code></pre>
<p>This command will produce <code>MGSCv37_gc_file.pytor</code> file that contains information about GC content in 100-base-pair bins.
For reference genomes where we have strict mask in the same format as <a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/working/20160622_genome_mask_GRCh38/">1000 Genomes Project strict mask</a>, we can create mask file using command:</p>
<pre><code>&gt; cnvpytor -root MGSCv37_mask_file.pytor -mask ~/hg19/mouse.strict_mask.whole_genome.fasta.gz -make_mask_file
</code></pre>
<p>If you do not have mask file, You can skip this step. Mask file contains information about regions of the genome that are more accessible to next generation sequencing methods using short reads.
CNVpytor uses P marked positions to filter SNP-s and read depth signal. If reference genome configuration does not contain mask file, CNVpytor will still be fully functional, apart from the filtering step.
You may also generate your own mask file by creating fasta file that contains character “P” if corresponding base pair passes the filter and any character different than “P” if not.</p>

</div>
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="calling-cnv-from-alignment-file-using-terminal.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
