<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Visualize CNVpytor data inside JBrowse | Copy Number Analysis using CNVpytor</title>
  <meta name="description" content="Chapter 7 Visualize CNVpytor data inside JBrowse | Copy Number Analysis using CNVpytor" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Visualize CNVpytor data inside JBrowse | Copy Number Analysis using CNVpytor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Chapter 7 Visualize CNVpytor data inside JBrowse | Copy Number Analysis using CNVpytor" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Visualize CNVpytor data inside JBrowse | Copy Number Analysis using CNVpytor" />
  
  <meta name="twitter:description" content="Chapter 7 Visualize CNVpytor data inside JBrowse | Copy Number Analysis using CNVpytor" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/dasl_favicon.ico" type="image/x-icon" />
<link rel="prev" href="visualization.html"/>
<link rel="next" href="about-the-authors.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Course</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html"><i class="fa fa-check"></i><b>2</b> Installation and Setting Reference genome</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#learning-objectives"><i class="fa fa-check"></i><b>2.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#libraries"><i class="fa fa-check"></i><b>2.2</b> Libraries</a></li>
<li class="chapter" data-level="2.3" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#installation"><i class="fa fa-check"></i><b>2.3</b> Installation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#install-by-cloning-from-github"><i class="fa fa-check"></i><b>2.3.1</b> Install by cloning from GitHub</a></li>
<li class="chapter" data-level="2.3.2" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#install-using-pip"><i class="fa fa-check"></i><b>2.3.2</b> Install using pip</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#steps-for-setting-reference-genome"><i class="fa fa-check"></i><b>2.4</b> Steps for setting reference genome</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="installation-and-setting-reference-genome.html"><a href="installation-and-setting-reference-genome.html#create-gc-and-mask-file-for-new-reference-genome"><i class="fa fa-check"></i><b>2.4.1</b> Create GC and mask file for new reference genome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html"><i class="fa fa-check"></i><b>3</b> Calling CNV from Alignment file using terminal</a>
<ul>
<li class="chapter" data-level="3.1" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="3.2" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html#calling-cnv-from-alignment-file"><i class="fa fa-check"></i><b>3.2</b> Calling CNV from Alignment file</a></li>
<li class="chapter" data-level="3.3" data-path="calling-cnv-from-alignment-file-using-terminal.html"><a href="calling-cnv-from-alignment-file-using-terminal.html#importing-and-using-variant-information"><i class="fa fa-check"></i><b>3.3</b> Importing and using variant information:</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-import.html"><a href="data-import.html"><i class="fa fa-check"></i><b>4</b> Data import</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-import.html"><a href="data-import.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.1</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html"><i class="fa fa-check"></i><b>5</b> Import read depth signal</a>
<ul>
<li class="chapter" data-level="5.1" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#check-reference-genome"><i class="fa fa-check"></i><b>5.1</b> check reference genome</a></li>
<li class="chapter" data-level="5.2" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#predicting-cnv-regions"><i class="fa fa-check"></i><b>5.2</b> Predicting CNV regions</a></li>
<li class="chapter" data-level="5.3" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#import-snp-data"><i class="fa fa-check"></i><b>5.3</b> Import SNP data</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#from-variant-file"><i class="fa fa-check"></i><b>5.3.1</b> From variant file</a></li>
<li class="chapter" data-level="5.3.2" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#using-snp-positions-from-variant-file-and-counts-from-alignment-file"><i class="fa fa-check"></i><b>5.3.2</b> Using SNP positions from variant file and counts from alignment file</a></li>
<li class="chapter" data-level="5.3.3" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#calculating-baf-histograms"><i class="fa fa-check"></i><b>5.3.3</b> Calculating BAF histograms</a></li>
<li class="chapter" data-level="5.3.4" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#predicting-cnv-regions-using-joint-caller-prototype"><i class="fa fa-check"></i><b>5.3.4</b> Predicting CNV regions using joint caller (prototype)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="import-read-depth-signal.html"><a href="import-read-depth-signal.html#genotyping-genomic-regions"><i class="fa fa-check"></i><b>5.4</b> Genotyping genomic regions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualization.html"><a href="visualization.html"><i class="fa fa-check"></i><b>6</b> Visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualization.html"><a href="visualization.html#learning-objectives-3"><i class="fa fa-check"></i><b>6.1</b> Learning Objectives</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="visualization.html"><a href="visualization.html#plot-from-command-line"><i class="fa fa-check"></i><b>6.1.1</b> Plot from command line</a></li>
<li class="chapter" data-level="6.1.2" data-path="visualization.html"><a href="visualization.html#plot-from-interactive-mode"><i class="fa fa-check"></i><b>6.1.2</b> Plot from interactive mode</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visualize-cnvpytor-data-inside-jbrowse.html"><a href="visualize-cnvpytor-data-inside-jbrowse.html"><i class="fa fa-check"></i><b>7</b> Visualize CNVpytor data inside JBrowse</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visualize-cnvpytor-data-inside-jbrowse.html"><a href="visualize-cnvpytor-data-inside-jbrowse.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="7.2" data-path="visualize-cnvpytor-data-inside-jbrowse.html"><a href="visualize-cnvpytor-data-inside-jbrowse.html#libraries-1"><i class="fa fa-check"></i><b>7.2</b> Libraries</a></li>
<li class="chapter" data-level="7.3" data-path="visualize-cnvpytor-data-inside-jbrowse.html"><a href="visualize-cnvpytor-data-inside-jbrowse.html#steps-to-process-cnvpytor-data-for-jbrowse-integration"><i class="fa fa-check"></i><b>7.3</b> Steps to process CNVpytor data for JBrowse integration</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by:</a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Copy Number Analysis using CNVpytor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=1.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/style.css" rel="stylesheet">
</head>



<div class="hero-image-container">
  <img class= "hero-image" src= "https://github.com/jhudsl/OTTR_Template/raw/main/assets/dasl_thin_main_image.png">
</div>
<div id="visualize-cnvpytor-data-inside-jbrowse" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> Visualize CNVpytor data inside JBrowse</h1>
<div id="learning-objectives-4" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Learning Objectives</h2>
<p>This chapter will cover:</p>
<ul>
<li>How to visualize CNVpytor data in JBrowse.</li>
</ul>
</div>
<div id="libraries-1" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Libraries</h2>
<pre><code>JBrowse version: https://github.com/GMOD/jbrowse/archive/1.16.6-release.tar.gz
 Plugins: 
 - multibigwig (https://github.com/elsiklab/multibigwig )
 - multiscalebigwig (https://github.com/cmdcolin/multiscalebigwig)

 **Note:** The JBrowse development version is required as integration of different jbrowse plugins are needed.</code></pre>
</div>
<div id="steps-to-process-cnvpytor-data-for-jbrowse-integration" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Steps to process CNVpytor data for JBrowse integration</h2>
<p>To generate CNVpytor file for JBrowse visualization:</p>
<pre><code>cnvpytor -root [pytor files] -export jbrowse [optional argument: output path]

Default export directory name: 
 - For single pytor file input:  jbrowse_[pytor file name]
 - For multiple pytor file input: cnvpytor_jbrowse_export</code></pre>
<p>The above command creates all the necessary files that are required to visualize the CNVpytor data.</p>
<p>To view CNVpytor file using JBrowse, users need to install JBrowse and required plugins (See JBrowse version and plugins section).
<code>http://localhost/jbrowse/?data=[export directory]</code></p>
<pre><code># Example usage
cnvpytor -root test.pytor -export jbrowse
http://localhost/jbrowse/?data=jbrowse_test</code></pre>
<div id="data-description" class="section level4" number="7.3.0.1">
<h4><span class="header-section-number">7.3.0.1</span> Data Description</h4>
<p>There are mainly two types of data CNVpytor processes. i.e.; Read depth data from alignment file and SNP data from variant file. Depending on the availability of these two input data, the export function works.</p>
<p>For Read depth data, it exports Raw segmented RD, GC corrected Raw Segmented RD, GC corrected RD partition, CNV calling using RD . All of these Read depth signals are plotted on top of each other on a single horizontal track using color gray, black, red and green respectively.
For SNP data, it exports Binned BAF, Likelihood of the binned BAF signals. These two signals are plotted on top of each other with gray and red color.</p>
<p align="center">
</p>
<table>
<thead>
<tr>
<th align="left">
Data
</th>
<th align="center">
Signal name with color on JBrowse
</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">
Read Depth (RD)
</td>
<td align="left">
Raw Segmented RD (Gray) <br>GC Corrected Raw Segmented RD (Black) <br> GC corrected RD partition (Red) <br> CNV call using RD signals (Green)
</td>
</tr>
<tr>
<td align="left">
SNP
</td>
<td align="left">
Binned BAF (Gray) <br> Likelihood of the Binned BAF (Red)
</td>
</tr>
</tbody>
</table>
<p>
</p>
<p>CNVpytor does the segmentation for all of the above data based on the user provided bin size. The <code>multiscalebigwig</code> provides the option to show the data based on the visualized area on the reference genome, which means if a user visualizes a small region of the genome it shows small bin data and vice versa.<br />
!</p>

</div>
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="visualization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="about-the-authors.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
